incubator/exporter-otlp/src/main/java/io/aklivity/zilla/runtime/exporter/otlp/internal/OltpExporterHandler.java

```diff
@@ -0,0 +1,147 @@
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *   https://www.aklivity.io/aklivity-community-license/
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.aklivity.zilla.runtime.exporter.otlp.internal;
+
+import static io.aklivity.zilla.runtime.exporter.otlp.internal.config.OtlpOptionsConfig.OtlpSignalsConfig.METRICS;
+
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+import java.util.function.LongFunction;
+
+import io.aklivity.zilla.runtime.engine.EngineContext;
+import io.aklivity.zilla.runtime.engine.config.AttributeConfig;
+import io.aklivity.zilla.runtime.engine.config.KindConfig;
+import io.aklivity.zilla.runtime.engine.exporter.ExporterHandler;
+import io.aklivity.zilla.runtime.engine.metrics.Collector;
+import io.aklivity.zilla.runtime.engine.metrics.reader.MetricsReader;
+import io.aklivity.zilla.runtime.exporter.otlp.internal.config.OtlpExporterConfig;
+import io.aklivity.zilla.runtime.exporter.otlp.internal.config.OtlpOptionsConfig;
+import io.aklivity.zilla.runtime.exporter.otlp.internal.serializer.OtlpMetricsSerializer;
+
+public class OltpExporterHandler implements ExporterHandler
+{
+    private static final String HTTP = "http";
+
+    private final long retryInterval;
+    private final Duration timeoutInterval;
+    private final long warningInterval;
+    private final EngineContext context;
+    private final Set<OtlpOptionsConfig.OtlpSignalsConfig> signals;
+    private final String protocol;
+    private final URI metricsEndpoint;
+    private final long interval;
+    private final Collector collector;
+    private final LongFunction<KindConfig> resolveKind;
+    private final List<AttributeConfig> attributes;
+    private final HttpClient httpClient;
+    private final Consumer<HttpResponse<String>> responseHandler;
+
+    private OtlpMetricsSerializer serializer;
+    private long nextAttempt;
+    private long lastSuccess;
+    private boolean warningLogged;
+    private CompletableFuture<HttpResponse<String>> response;
+
+    public OltpExporterHandler(
+        OltpConfiguration config,
+        EngineContext context,
+        OtlpExporterConfig exporter,
+        Collector collector,
+        LongFunction<KindConfig> resolveKind,
+        List<AttributeConfig> attributes)
+    {
+        this.retryInterval = config.retryInterval();
+        this.timeoutInterval = config.timeoutInterval();
```

Comment:
This is a good solution, to support timeout of the response in case it takes too long.

However, I think we want to make sure `timeout.interval` cannot be set higher than `retry.interval` so that only one request-response is in flight at once.

Suggest `this.timeoutInterval = Math.min(config.timeoutInterval(), config.retryInterval());`

Reply:
I don't think that's necessary, sending a new request is now inside the `if (response == null || response.isDone())` block. So a new request will not be sent as long as the latest request is still in-flight.

If the `timeout.interval` is set to higher that the `retry.interval`, the loop works as expected, waiting for the request to time out, and staring the next request right then.